x/common/dec.go

```diff
@@ -0,0 +1,59 @@
+package common
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// Computes the square root of the input decimal using its underlying big.Int
+// The big.Int.Sqrt method is part of the standard library and thoroughly tested.
+//   - NOTE, MustSqrtDec panics if it is called on a negative number, similar to the
+//     sdk.NewCoin and SqrtBigInt functions.
+func MustSqrtDec(dec sdk.Dec) sdk.Dec {
+	sqrtBigInt := SqrtBigInt(dec.BigInt())
+	precision := sdk.NewDecFromBigInt(PRECISION_MULT)
+	return sdk.NewDecFromBigInt(sqrtBigInt).Quo(precision)
+}
+
+func SqrtDec(dec sdk.Dec) (sdk.Dec, error) {
+	var sqrtDec sdk.Dec
+	var panicErr error = TryCatch(func() {
+		sqrtDec = MustSqrtDec(dec)
+	})()
+	return sqrtDec, panicErr
+}
+
+// SqrtBigInt returns the square root of the input.
+//   - NOTE: SqrtBigInt panics if it is called on a negative number because it uses
+//     the `big.Int.Sqrt` from the "math/big" package.
+func SqrtBigInt(i *big.Int) *big.Int {
```

Comment:
This should be `MustSqrtBigInt`, and `SqrtBigInt` should have an error in the return